WireDomain/Sources/WireDomain/Notifications/Builders/ConversationNotificationBuilder.swift

```diff
@@ -0,0 +1,141 @@
+            id: event.senderID.uuid,
+            domain: event.senderID.domain
+        )
+
+        let conversation = await conversationLocalStore.fetchOrCreateConversation(
+            id: event.conversationID.uuid,
+            domain: event.conversationID.domain
+        )
+
+        let conversationMutedMessages = await conversationLocalStore.conversationMutedMessageTypesIncludingAvailability(
+            conversation
+        )
+
+        let isConversationMuted = conversationMutedMessages != .none
+
+        let eventTimeStamp = event.timestamp
+        let lastReadTimestamp = await conversationLocalStore.lastReadServerTimestamp(conversation)
+
+        self.context = Context(
+            senderID: event.senderID,
+            conversationID: event.conversationID,
+            isSelfUser: isSelfUser == true,
+            isConversationMuted: isConversationMuted,
+            eventTimeStamp: eventTimeStamp,
+            lastReadTimestamp: lastReadTimestamp
+        )
+    }
+
+    func buildContent() async -> UNMutableNotificationContent {
+        let builder: NotificationBuilder
+
+        switch event {
+        case let .mlsMessageAdd(mlsMessageEvent):
+            let decryptedMessage = mlsMessageEvent.decryptedMessages.first?.message
+
+            guard let decryptedMessage,
+                  let (genericMessage, _) = ProtobufMessageHelper.getProtobufMessage(
+                      from: decryptedMessage
+                  ) else {
+                return UNMutableNotificationContent()
+            }
+
+            builder = await NewMessageNotificationBuilder(
+                message: genericMessage,
+                conversationID: mlsMessageEvent.conversationID,
+                senderID: mlsMessageEvent.senderID
+            )
+
+        case let .proteusMessageAdd(proteusMessageEvent):
+            let decryptedMessage = proteusMessageEvent.message.decryptedMessage
+            let externalEncryptedMessage = proteusMessageEvent.externalData?.encryptedMessage
+
+            guard let decryptedMessage,
+                  let (genericMessage, _) = ProtobufMessageHelper.getProtobufMessage(
+                      from: decryptedMessage,
+                      externalData: externalEncryptedMessage
+                  ) else {
+                return UNMutableNotificationContent()
+            }
+
+            builder = await NewMessageNotificationBuilder(
+                message: genericMessage,
+                conversationID: proteusMessageEvent.conversationID,
+                senderID: proteusMessageEvent.senderID
+            )
+
+        default: // TODO: [WPB-11175] - Generate notifications for other events
+            return UNMutableNotificationContent()
+        }
+
+        guard await builder.shouldBuildNotification() else {
+            return UNMutableNotificationContent()
+        }
+
+        return await builder.buildContent()
+    }
+
+    func shouldBuildNotification() async -> Bool {
```

Comment:
I suppose this is to be called at a higher level? It's a little confusing to have this method, plus the sub-builder method of the same name.